src/domain/orders/edit/modal.tsx

```diff
@@ -367,43 +365,41 @@ type OrderEditModalContainerProps = {
   order: Order
 }
 
+let isRequestRunningFlag = false
+
 function OrderEditModalContainer(props: OrderEditModalContainerProps) {
   const { order } = props
   const notification = useNotification()
 
-  const {
-    hideModal,
-    orderEdits,
-    activeOrderEditId,
-    setActiveOrderEdit,
-  } = useContext(OrderEditContext)
+  const { hideModal, orderEdits, activeOrderEditId, setActiveOrderEdit } =
+    useContext(OrderEditContext)
 
-  const { mutate: createOrderEdit } = useAdminCreateOrderEdit()
+  const { mutateAsync: createOrderEdit } = useAdminCreateOrderEdit()
 
   const orderEdit = orderEdits?.find((oe) => oe.id === activeOrderEditId)
 
   useEffect(() => {
-    if (activeOrderEditId) {
+    if (activeOrderEditId || isRequestRunningFlag) {
       return
     }
 
-    createOrderEdit(
-      { order_id: order.id },
-      {
-        onSuccess: ({ order_edit }) => {
-          setActiveOrderEdit(order_edit.id)
-        },
-        onError: () => {
-          notification(
-            "Error",
-            "There is already active order edit on this order",
-            "error"
-          )
-          hideModal()
-        },
+    isRequestRunningFlag = true
+    ;(async function () {
+      try {
+        const { order_edit } = await createOrderEdit({ order_id: order.id })
+        setActiveOrderEdit(order_edit.id)
+      } catch (e) {
+        notification(
+          "Error",
+          "There is already active order edit on this order",
```

Comment:
Suggestion : there is already an active…

Reply:
Question : can this happen? Since we are checking if the order edit id exists and return in that case?

Reply:
yes - `activeOrderEditId` represents OE that the user selected for editing(to be showed in the modal), while failed request means there is already an edit in the DB that is created/requested